Aptoide/bds-sdk
# feature/APPC-1085-new-wallet-installation-dialog

**What does this PR do?**

* Add DialogWalletInstall  class component to android-appcoins-billing project
* Add android-appcoins-billing as api dependency of appcoins-ads in order to use DialogWalletInstall
* Change implementation at promptToInstallWallet at WalletUtils to use DialogWalletInstall
* Add i18n strings according to APPC-1061

**Database changed?**
 No

**Where should the reviewer start?**

* com.appcoins.sdk.billing.wallet.DialogWalletInstall.java
* android-appcoins-billing/res/layout/wallet_install_dialog.xml

* android-appcoins-billing/res/drawable/dialog_wallet_install_download_button_background.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_empty_image.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_parent_background.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_shape_color_normal.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_shape_color_pressed.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_graphic.xml
* android-appcoins-billing/res/drawable/dialog_wallet_install_icon.xml
* android-appcoins-billing/res/values-pt-rBR/string.xml
* android-appcoins-billing/res/values-es/string.xml
* android-appcoins-billing/res/values-en/string.xml
* android-appcoins-billing/res/values-de/string.xml

* com.appcoins.sdk.billing.helpers.WalletUtils.java
* com.asf.appcoins.sdk.ads.poa.manager.WalletUtils.java
* appcoins-ads/build.gradle


**How should this be manually tested?**

  * Open trivialdrivesample(toolbox) and see wallet install popup show up from com.asf.appcoins.sdk.ads.poa.manager
  * Click in buy flow button at  trivialdrivesample(toolbox) and see wallet install popup show up from com.appcoins.sdk.billing.helpers

**What are the relevant tickets?**
[APPC-1085](https://aptoide.atlassian.net/browse/APPC-1085)

**Questions:**
1. Add all drawables related to new DialogWalletInstall  some mocks related to graphic and icon , since it will be added afterward by apk-migrator, get those from ws2 api  and then hasImage flag will be also set later. 
2. We can make DialogWalletInstall component read this flag always from a bool.xml which gonna be written by apk-migrator and simplify the .with() factory method

**Code Review Checklist**
- [N/A] Architecture
- [N/A] Database changed?
- [N/A] If yes - Database migration?
- [N/A] Remove comments & unused code & forgotten testing Logs
- [N/A] Functional QA tests pass
